src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,142 @@
+import json from "./AllowedVocab.json";
+import Issue from "./Issue";
+const punctuation = new RegExp(/[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/g);
+const space = new RegExp(/\s+/g);
+const allExceptSpace = new RegExp(/^(?!\s*$).+/g);
+const allowedWords = json;
+
+/**
+ * (input:) @param text
+ * This method takes the user input as a parameter.
+ * At first it prepocesses @param text by calling
+ * @function preprocessText thus calls the two
+ * functions @function eliminatePunctuation and @function splitIntoWords.
+ * For their functionality see the corresponding comments.
+ * @function checkInput then filters out all wrong @param word
+ * from @param text and puts them all into @param allWrongWords,
+ * which is printed in the console log by calling @function logInvalidWords.
+ * This is for debugging purposes and should later be changed to
+ * be displayed to the user. For further postprocessing all wrong
+ * words are collected in a @returns @param invalidWords of
+ * @type string array.
+ */
+
+export function checkInput(text: string) {
+		const preprocessedText: string[] = preprocessText(text);
+		const invalidWords = preprocessedText.filter(word => !allowedWords.includes(word));
+		// TODO: Make logInvalidWords dependent on this posOfInvalidWords.
+		const mapOfInvalidWords = createMapOfInvalidWords(text, invalidWords);
```

Comment:
You already searched through the text to find your wrong words, now you search through it again? I know that O(n)=O(2n), but still, this is just unnecessary ;) You can just find the positions during your first go.